source/common/stats/utility.cc

```diff
@@ -26,6 +26,25 @@ std::string Utility::sanitizeStatsName(absl::string_view name) {
                                    });
 }
 
+absl::string_view Utility::sanitizeStatsName(absl::string_view name, std::string& buffer) {
+  if (absl::EndsWith(name, ".")) {
+    name.remove_suffix(1);
+  }
+  if (absl::StartsWith(name, ".")) {
+    name.remove_prefix(1);
+  }
+  // Check if the name contains a '\0' character.
```

Comment:
Document the motivation here, e.g. s/Check/Skip the string-copy in StrReplaceAll if there is no embedded nul character.